### PR TITLE
Upstream a few things from LDC to reduce the diff

### PR DESCRIPTION
**Pull request by kinke**

### CODE REVIEW

File: src/rt/sections_elf_shared.d [hunk 1]
```diff
@@ -95,6 +95,7 @@ struct DSO
         return _moduleGroup;
     }
 
+    version (DigitalMars)
     @property immutable(FuncTable)[] ehTables() const nothrow @nogc
```
Comment (kinke) [inline src/rt/sections_elf_shared.d:99]: @ibuclaw: I assume gdc doesn't feature this method either.

File: src/rt/sections_darwin_64.d [hunk 2]
```diff
@@ -176,3 +178,28 @@ const(section_64)[] sections(const segment_command_64* segment) pure nothrow @no
     const firstSection = cast(const(section_64)*)(cast(ubyte*) segment + size);
     return firstSection[0 .. segment.nsects];
 }
+
+/// Identifies a section via segment and section names.
+struct SectionRef
+{
+    immutable(char)* seg;
+    immutable(char)* sect;
+}
```
Comment (Geod24) [inline src/rt/sections_darwin_64.d:187]: Personally I'd rather have them as string, as previously. Why the change ?

File: src/rt/sections_darwin_64.d [hunk 3]
```diff
@@ -176,3 +178,28 @@ const(section_64)[] sections(const segment_command_64* segment) pure nothrow @no
...
+
+/// The data sections of an image (executable or shared library).
+static immutable SectionRef[] dataSections = [{SEG_DATA, SECT_DATA},
```
Comment (Geod24) [inline src/rt/sections_darwin_64.d:190]: `static` here is redundant

File: src/rt/sections_darwin_64.d (hunk 2 above)
Comment (kinke) [reply src/rt/sections_darwin_64.d:187]: Because the single usage - for `getSection()` below - requires null-terminated char* pointers.

Comment (kinke) [reply src/rt/sections_darwin_64.d]: I've refactored this & amended the last commit.
